frame/timestamp/src/lib.rs

```diff
@@ -159,9 +159,9 @@ decl_module! {
 		/// The dispatch origin for this call must be `Inherent`.
 		///
 		/// # <weight>
-		/// - `O(T)` where `T` complexity of `on_timestamp_set`
+		/// - `O(1)` (Note that implementations of `OnTimestampSet` must also be `O(1)`)
```

Comment:
Even if OnTimestampSet is `O(1)` the benchmark might miss some read/writes and base cost.

benchmark only call `on_timestamp_set(100)`, but maybe `on_timestamp_set(100)` is 1 storage read whereas `on_timestamp_set(200)` will trigger some storage writes because some period are over or something.

like if a module do 100 storage writes every minutes. (obviously in this case such a pallet should be refactored to use on_initialize, but my point is that such implementation is `O(1)`).

maybe we can document that on_timestamp_set should always cost its worst case, although this is an annoying constraint.
or maybe I miss something in the picture here.

(note aside, I don't think this is that much of an issue here, I just wonder about general usage of handler)

Reply:
Maybe O(1) means constant time right? i.e. this function **should** always have the same weight independent of the input.

Reply:
I think constant time means there is an upper bound independant from input.
the following code is constant time:
```rust
fn foo(a: u32) {
  if a > 10 {
    do 2 writes
  }
}
```
https://en.wikipedia.org/wiki/Time_complexity#Constant_time

Reply:
Yeah, I am not sure what else I can do in this case then.